### ISSUE TITLE
Consider masking email addresses in logs

### ISSUE THREAD

**Issue by vierbergenlars** [iRail/iRail#396]
The best practices for API usage at https://docs.irail.be/#header-best-practices-when-using-the-api tell to use a contact email address in the user-agent header, so you can contact developers in case of issues.

However, the provided user-agent is also exposed in the public access logs (https://docs.irail.be/#log-endpoint-logs-api-get).

Today, I started receiving spam on the unique email address I provided in the user-agent (irail@vbgn.be), so at least one spammer is now parsing these logs to find email addresses.

Would you consider masking email addresses in the public API logs? That way, external analytics on applications using irail are still possible, but developer email addresses are protected from spammers.

**Comment by Bertware** [comment]
I agree on this one, there shouldn't be any negative consequences for those who provide a contact mail address. How would you mask them? Stripping them completely, or using something like *****@vgbn.be?

**Comment by Bertware** [comment]
Email addresses are now obfuscated in public log files:

`pyRail (t**************@g****.com)`

It'd be easiest to run a similar script tomorrow on historical data. If we'd run it today, we'd still have to run it one more time tomorrow for todays log file.

**Comment by vierbergenlars** [comment]
Thanks for the very fast response!

The way you obfuscate email addresses looks ideal.
Keeping the first letter and the length of the email address should disambiguate email addresses enough for most analytics purposes, but makes it impossible to recover email addresses from it.

As for the historical data, I don't think waiting for another day so you only need to run a script once is that bad. The current spammer probably already has stored the email addresses, and chances that an other spammer will find it now are quite slim.

**Comment by pietercolpaert** [comment]
> @pietercolpaert thoughts?

Perfect! Thanks for the quick response!
